crypto/keys/src/ristretto.rs

```diff
@@ -75,6 +82,37 @@ impl RistrettoPrivate {
     pub fn to_bytes(&self) -> [u8; 32] {
         *self.0.as_bytes()
     }
+
+    /// Sign the given bytes using a deterministic scheme based on Schnorrkel.
+    pub fn sign_schnorrkel(&self, context: &[u8], message: &[u8]) -> SchnorrkelSignature {
+        // Create a deterministic nonce using a merlin transcript. See this crate's README
+        // for a security statement.
+        let nonce = {
+            let mut transcript = MerlinTranscript::new(b"SigningNonce");
+            transcript.append_message(b"context", &context);
+            transcript.append_message(b"private", &self.to_bytes());
+            transcript.append_message(b"message", &message);
+            let mut nonce = [0u8; 32];
+            transcript.challenge_bytes(b"nonce", &mut nonce);
+            nonce
+        };
+
+        // Construct a Schnorrkel SecretKey object from ourselves, and our nonce value
+        let mut secret_bytes = [0u8; 64];
+        secret_bytes[0..32].copy_from_slice(&self.to_bytes());
+        secret_bytes[32..64].copy_from_slice(&nonce);
+        let secret_key = SchnorrkelPrivate::from_bytes(&secret_bytes).unwrap();
+        let keypair = secret_key.to_keypair();
+
+        // SigningContext provides domain separation for signature
+        let mut t = MerlinTranscript::new(b"SigningContext");
+        t.append_message(b"", context);
```

Comment:
Curious why the label is empty here (but not on the nonce for example)

Reply:
This was what the original `mc-crypto-sig` crate contained, I tried to avoid touching signature construction more than replacing, e.g. `private_key` with `self`.